### ISSUE TITLE
--experimental_remote_downloader_config not working

### ISSUE THREAD

**Issue by pilkjaer** [bazelbuild/bazel#12338]
### Description of the problem / feature request:

Mechanism to configure the downloader is not working. Related to https://github.com/bazelbuild/bazel/commit/9d2946440a43c2adc6b3cdf09739a6dbf5e29abf

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

$ bazel build --experimental_remote_downloader_config .bazel_download_config ...
or
$ bazel build --experimental_remote_downloader_config=.bazel_download_config ...

### What operating system are you running Bazel on?

Red Hat Enterprise Linux Server release 7.8 (Maipo)

### What's the output of `bazel info release`?

release 3.7.0

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

I run 8cffdf1b1db2c22dfc35cb2c8c7c9e2ff1932cad commit which is 3.7.0 release commit.

###  Have you found anything relevant by searching the web?

No

### Any other information, logs, or outputs that you want to share?

$ bazel build --experimental_remote_downloader_config=.bazel_download_config ...
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
ERROR: --experimental_remote_downloader_config=.bazel_download_config :: Unrecognized option: --experimental_remote_downloader_config=.bazel_download_config

$ bazel build --experimental_remote_downloader_config .bazel_download_config ...
ERROR: --experimental_remote_downloader_config :: Unrecognized option: --experimental_remote_downloader_config

$ bazel help build | grep experimental_remote_downloader_config
<no results>

**Comment by coeuvre** [comment]
The flag is renamed as `--experimental_downloader_config`.

**Comment by pilkjaer** [comment]
@coeuvre  thank you for the update. Are you going to update Bazel documentation mentioning correct flag name as well as how to use it?
I tried to use the feature and I got following text in the console that seems rather wired:

INFO: Repository 'rules_cc' used the following cache hits instead of downloading the corresponding file.
 * Hash '1d4dbbd1e1e9b57d40bb0ade51c9e882da7658d5bfbf22bbd15b68e7879d761f' for http://nonexistent.example.org/cacheprobe
If the definition of 'rules_cc' was updated, verify that the hashes were also updated.

**Comment by pilkjaer** [comment]
I figured out the problem. In the https://github.com/bazelbuild/bazel/commit/9d2946440a43c2adc6b3cdf09739a6dbf5e29abf in the commit message mentioned "This adds a `--experimental_remote_downloader_config` flag" but in the src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java file it's name = "experimental_downloader_config"

It would be nice that in the future reviewers and committers will be more thorough in their work to avoid confusion. Thank you, @coeuvre , for support!

**Comment by coeuvre** [comment]
> Are you going to update Bazel documentation mentioning correct flag name as well as how to use it?

I am happy to do that. You can keep is issue opened until the doc is updated :=)

> It would be nice that in the future reviewers and committers will be more thorough in their work to avoid confusion.

Sorry for the confusion and thanks for reporting this. Actually, it was me reviewed the PR for this feature. Initially, the flag is named as `--experimental_remote_downloader_config` and I thought it could be confused with remote options, so committer renamed it to `--experimental_downloader_config` before merging but forgot to update the commit message.

**Comment by pilkjaer** [comment]
I have tried out the feature and I have some observations that can be considered being bugs or implementation details.

Background
we have a custom Bazel installation that we want to limit where it can download files from. We are using --distdir flag to provide a path to a folder that contains all the files that are needed by the Bazel itself. This is supposed to make Bazel installation "self-contained" and only files that need to be downloaded are user-created ones.
We have a similar implementation that we have implemented ourselves and we have high expectation on this feature so that we can start using it instead of our custom solution.
My use case for that feature is that I want to be able to provide "allow" access to our custom storage and to "block" access to everything else.

To begin with, I was expecting to get an error message with a clear error stating that you are not allowed to download files from a specific place. Currently when providing only "block *" in the configuration file there is:
a) no error being generated for accesses outside of allowed range (maybe it should be another configurable flag if you don't want to enforce it?)
b) INFO text in the console contains just "INFO: Rewritten [<path_to_a_file>] as []" without any details why this is being done which might make troubleshooting for the end user problematic
c) Error message generated in the end is unclear: "ERROR: no such package '@bazel_stardoc//': java.io.IOException: Cache miss and no url specified"

So in general there is no connection between behaviour observed by the user in the console and feature doing the modifications in the background which can cause confusion for the end user and complicate troubleshooting.

I have tried to combine "block *" with "allow <my_hostname>" and there is no way to make it work. I tried to change places for them but files are still not being downloaded from my custom hostname. I see this as a bug.

I have tried only having "allow <my_hostname>" in the file but then everything else is being downloaded as well. So flag/file has basically no effect. I don't know if this is intended behaviour. I was expecting "block *" to be some kind of implicit internal rule as soons you provide --experimental_downloader_config.

There seems to be no way to include files in cases when you want to split or re-use configuration. It would be nice to have that feature as well.
